protocol/contracts/reserve/src/contract.rs

```diff
@@ -0,0 +1,118 @@
+use std::ops::DerefMut;
+
+use access_control::SingleUserAccess;
+use finance::coin::Coin;
+use platform::{
+    bank::{self, BankAccount, BankAccountView},
+    batch::Batch,
+    contract, error as platform_error,
+    message::Response as PlatformResponse,
+    response,
+};
+use sdk::{
+    cosmwasm_ext::Response as CwResponse,
+    cosmwasm_std::{self, entry_point, Binary, Deps, DepsMut, Env, MessageInfo},
+};
+use versioning::{package_version, version, SemVer, Version, VersionSegment};
+
+use crate::{
+    api::{ConfigResponse, ExecuteMsg, InstantiateMsg, LpnCurrency, MigrateMsg, QueryMsg},
+    error::{ContractError, ContractResult},
+    state::Config,
+};
+
+// const CONTRACT_STORAGE_VERSION_FROM: VersionSegment = 0;
+const CONTRACT_STORAGE_VERSION: VersionSegment = 0;
+const PACKAGE_VERSION: SemVer = package_version!();
+const CONTRACT_VERSION: Version = version!(CONTRACT_STORAGE_VERSION, PACKAGE_VERSION);
+
+#[entry_point]
+pub fn instantiate(
+    mut deps: DepsMut<'_>,
+    _env: Env,
+    _info: MessageInfo,
+    new_reserve: InstantiateMsg,
+) -> ContractResult<CwResponse> {
+    deps.api
+        .addr_validate(new_reserve.lease_code_admin.as_str())
+        .map_err(Into::<ContractError>::into)
+        .and_then(|_| {
+            SingleUserAccess::new(
+                deps.storage.deref_mut(),
+                crate::access_control::LEASE_CODE_ADMIN_KEY,
+            )
+            .grant_to(&new_reserve.lease_code_admin)
+            .map_err(Into::into)
+        })
+        .and_then(|()| versioning::initialize(deps.storage, CONTRACT_VERSION).map_err(Into::into))
+        .and_then(|()| Config::from(new_reserve).store(deps.storage))
+        .map(|()| response::empty_response())
+        .or_else(|err| platform_error::log(err, deps.api))
+}
+
+#[entry_point]
+pub fn migrate(deps: DepsMut<'_>, __env: Env, _msg: MigrateMsg) -> ContractResult<CwResponse> {
+    versioning::update_software(deps.storage, CONTRACT_VERSION, Into::into)
+        .and_then(response::response)
+        .or_else(|err| platform_error::log(err, deps.api))
```

Comment:
Since `platform::error::log` is infallible, I think the behavior we desire is `Result::{map|inspect}_err`. This goes in a very roundabout way about it when internally it's equivalent to `|error| Err(error)`.

Reply:
it's [a known flaw](https://github.com/nolus-protocol/nolus-money-market/blob/93c9485b4345551d55fa46472504b212db9e46ad/platform/packages/platform/src/error.rs#L80) which will be fixed once the Result::inspect_err get stabilized